File: src/main/java/com/example/storage/repository/EquipmentRepository.java
package com.example.storage.repository;

import com.example.storage.entity.Equipment;
import com.example.storage.entity.EquipmentStatus;
import org.springframework.data.jpa.repository.EntityGraph;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface EquipmentRepository extends JpaRepository<Equipment, Long> {
    @EntityGraph(attributePaths = {"category"})
    List<Equipment> findAllByStatus(EquipmentStatus status);
}

File: src/main/kotlin/ru/nto/storage/repository/EquipmentRepository.kt
package ru.nto.storage.repository

import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import ru.nto.storage.entity.Equipment
import ru.nto.storage.entity.EquipmentStatus

interface EquipmentRepository : JpaRepository<Equipment, Long> {
    @EntityGraph(attributePaths = ["category"])
    fun findAllByStatus(status: EquipmentStatus): List<Equipment>
}
